### PR TITLE
Call tslint and tsfmt command instead of shell commands

### PR DESCRIPTION
**Pull request by GermainBergeron**
Replace usage of shelljs with tslint and tsfmt
Remove shelljs from dependencies
Add .idea to .gitignore

### CODE REVIEW

**Review by gdostie — APPROVED** [review]

**Review by stephr0y — APPROVED** [review]

**Review by MichelML — APPROVED** [review]

File: tsjs.js [hunk 1]
```diff
@@ -33,12 +33,33 @@ try {
         fs.writeFileSync(tempTsConfigFile, JSON.stringify(tsConfigLint));
         tsConfigFile = tempTsConfigFile;
     }
+
+    const program = Linter.createProgram(tsConfigFile, ".");
+    const linter = new Linter({fix: true}, program);
+
```
Comment (MichelML) [inline tsjs.js]:
```
console.log(`Starting tslint with the following configuration:  
tsconfig: ${tsConfigFile}  
tslint: tsjs configuration
excluded files:
${[...tsConfigLint.exclude, ...excludeOptionArray].join('\n')}`);  
```

File: tsjs.js [hunk 2]
```diff
@@ -33,12 +33,33 @@ try {
...
+    const program = Linter.createProgram(tsConfigFile, ".");
+    const linter = new Linter({fix: true}, program);
+
+    const files = Linter.getFileNames(program);
+    files.forEach((file) => {
+        const fileContents = program.getSourceFile(file).getFullText();
+        const configuration = Configuration.findConfiguration(tempTsLintFile, file).results;
+        linter.lint(file, fileContents, configuration);
+    });
+
```
Comment (MichelML) [inline tsjs.js:56]:
```
console.log(`Starting tsfmt with the following configuration:  
tslint: tsjs configuration  
tsfmt: tsjs configuration  
tsconfig: ${tsConfigFile}  
excluded files:
${[...tsConfigLint.exclude, ...excludeOptionArray].join('\n')}`);  
```

File: tsjs.js [hunk 3]
```diff
@@ -33,12 +33,33 @@ try {
...
+        tslintFile: tempTsLintFile,
+        editorconfig: false,
+        vscode: false,
+        vscodeFile: null,
+        tsfmt: true,
+        tsfmtFile: tempTsfmtFile,
+    })
+        .then(() => console.log('done'));
+
 } catch (e) {
```
Comment (MichelML) [inline tsjs.js]:
```   
console.log('\nSomething went wrong while running tsjs. See error below and additional logs above for details.\n');   
```

File: tsjs.js [hunk 4]
```diff
@@ -33,12 +33,33 @@ try {
...
+        tsconfigFile: tsConfigFile,
+        tslint: true,
+        tslintFile: tempTsLintFile,
+        editorconfig: false,
+        vscode: false,
+        vscodeFile: null,
+        tsfmt: true,
+        tsfmtFile: tempTsfmtFile,
+    })
+        .then(() => console.log('done'));
```
Comment (MichelML) [inline tsjs.js]: `Tsjs linting and formatting completed successfully. See additional logs above for details.`

**Comment by MichelML** [comment]
screen shot 2018-08-08 at 10 59 46 am

@GermainBergeron this is what it looks like

**Pull request merged**